client/state/data-layer/wpcom-http/actions.js

```diff
@@ -30,15 +40,49 @@ export const http = ( {
 	onFailure,
 	onProgress,
 	...options,
-}, action = null ) => ( {
-	type: WPCOM_HTTP_REQUEST,
-	body,
-	method,
-	path,
-	query: { ...query, apiVersion },
-	formData,
-	onSuccess: onSuccess || action,
-	onFailure: onFailure || action,
-	onProgress: onProgress || action,
-	options,
-} );
+}, action = null ) => {
+	const actionProperties = {
+		onSuccess: onSuccess || action,
+		onFailure: onFailure || action,
+		onProgress: onProgress || action,
+		query: { ...query }
+	};
+
+	// * check parameters *
+
+	// `path` is not optional and it is a string
+	// `method` is not optional and its value can take either 'GET' or 'POST`
+	if (
+		( ! path || typeof path !== 'string' ) ||
+		( ! method || ( method !== 'GET' && method !== 'POST' ) )
+	) {
+		return badRequest();
+	}
+
+	// - One of `apiVersion` of `apiNamespace` should be defined
+	// - `apiVersion` and `apiNamespace` cannot be defined simultaneously
+	if (
+		( typeof apiVersion === 'undefined' && typeof apiNamespace === 'undefined' ) ||
+		( typeof apiVersion !== 'undefined' && typeof apiNamespace !== 'undefined' )
+	) {
+		return badRequest();
+	}
+
+	// - `apiVersion` must have the `v1, v2, ...` shape
+	if ( typeof apiVersion !== 'undefined' && ( /^v\d+/ ).test( apiVersion ) ) {
+		actionProperties.query.apiVersion = apiVersion;
```

Comment:
in the end, let's merge these in in one declarative statement instead of directly mutating the query object - will make for significantly easier code to follow and maintain